Cython/Shadow.pyi

```diff
@@ -100,3 +101,34 @@ class typedef(CythonType, Generic[_T]):
 #    def __init__(self) -> None: ...
 
 #def fused_type(*args: Tuple[_T]) -> Type[FusedType[_T]]: ...
+
+def typeof(arg: Any) -> str: ...
+
+_C = TypeVar('_C', bound='Callable')
+
+class _EmptyDecoratorAndManager(object):
+    def __call__(self, x: _C) -> _C: ...
+
+    def __enter__(self) -> None: ...
+
+    def __exit__(
+        self,
+        exc_type: Optional[type[BaseException]],
+        exc: Optional[BaseException],
+        tb: Optional[TracebackType]
+    ) -> None: ...
+
+cclass = ccall = cfunc = inline = _EmptyDecoratorAndManager()
+
+@overload
+def declare(
+    t: Optional[Callable[..., _T]] = None,
+    value: Any = ...,
+    *,
+    visibility: str = ...
+) -> _T:
+    ...
```

Comment:
I think it'll be better to have it like this:
```python
visibility: str = ...) -> _T: ...
```

Reply:
It is already expanded to a hanging indent, I think it looks worse like this when you put the whole function in perspective
```py
@overload
def declare(
    t: Optional[Callable[..., _T]] = None,
    value: Any = ...,
    *,
    visibility: str = ...) -> _T: ...
```
If we use a visual indent it will probably work
```py
@overload
def declare(t: Optional[Callable[..., _T]] = None,
            value: Any = ...,
            *,
            visibility: str = ...) -> _T: ...
```
But in general visual indents like this look worse.